### PR TITLE
Reduce tracing default to avoid overload

### PR DESCRIPTION
**Pull request by costinm**
- reduce default tracing rate from 100% to 1% - ingress and sidecars
    are affected if jaeger is overloaded.
    
    - allow image override for ingress
    
    - for the test cluster, add a second gateway pinned to 1.0.0 ( will
    update it to 1.0.1 after release ), so we can test upgrade and compare
    behavior.

### CODE REVIEW

**Comment by douglas-reid** [comment]
What is needed to ensure that the demo yaml gets 100% ?

File: install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml [hunk 1]
```diff
@@ -31,7 +31,11 @@ spec:
 {{- end }}
       containers:
         - name: istio-proxy
+{{- if contains "/" (toString $spec.image) }}
+          image: "{{ $spec.image }}"
+{{- else }}
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: why?

Comment (rshriram) [reply install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: oh ignore

File: install/kubernetes/helm/istio/values.yaml [hunk 2]
```diff
@@ -372,7 +372,8 @@ pilot:
   autoscaleMax: 5
   image: pilot
   sidecar: true
-  traceSampling: 100.0
+  # Default value for production must be low, otherwise ingress may crash.
+  traceSampling: 1.0
```
Comment (rshriram) [inline install/kubernetes/helm/istio/values.yaml]: set to 100 for istio-demo.. else tasks wont work

Comment (linsun) [reply install/kubernetes/helm/istio/values.yaml]: +1 on setting to 100 for demo :)

**Comment by douglas-reid** [comment]
Is this PR still needed? Are we moving forward with it?

**Comment by mandarjog** [comment]
@costinm I am closing this PR.
Trace sampling went into another PR.

**Pull request closed without merging**